stablehlo/reference/InterpreterOps.td

```diff
@@ -80,6 +80,29 @@ def Interpreter_RunParallelOp : Op<Interpreter_Dialect, "run_parallel", []> {
   let hasVerifier = 1;
 }
 
+def Interpreter_PrintOp : Op<Interpreter_Dialect, "print",
+    [SameOperandsAndResultType]> {
+  let summary = "Print operation";
+  let arguments = (ins
+    HLO_Tensor:$operand
```

Comment:
Why do we need an `in` argument here? Is the reason to create a dummy daa-dep so that the op cannot be optimized out, then we can use some side-affect like  `MemWrite`.

Reply:
Let's avoid outputs? Model it after `print` functions in other programming languages

Reply:
ref: https://github.com/llvm/llvm-project/blob/d2498afccb04c0f09b05827b6b9c1c6c181a4f2b/mlir/include/mlir/Dialect/Vector/IR/VectorOps.td#L2613

Reply:
By default not providing any "side effects" it assumes it has one. +1 to removing output. I was originally using it to get the SSA name, but I can get that via the operand instead.

Reply:
Thank!
How about we have the intent cleaner by specifying the exact side affect (in cases we have that knowledge) enabling the compiler to avoiding pessimistic assumptions.

Reply:
I think it's fine the way it is currently -- the purpose of this op is to inspect a given module and it isn't designed to run with other passes.